src/tasks.test.js

```diff
@@ -64,3 +64,44 @@ describe('add and remove', () => {
     expect(todoList.tasksArr).toHaveLength(2);
   });
 });
+
+describe('Edit test', () => {
+  test('Editing', () => {
+    const todoList = new TasksBluePrint();
+    const newTodo3 = {
+      id: '3',
+      description: 'task33',
+      completed: false,
+      index: 3,
+    };
+    todoList.add(newTodo3);
+    todoList.update(newTodo3.id, 'asd');
+    expect(todoList.tasksArr[2].description).toBe('asd');
+    expect(todoList.tasksArr).toHaveLength(3);
+  });
+});
+
+describe('complete test', () => {
+  test(' updating an item completed status', () => {
+    const todoList = new TasksBluePrint();
+    const newTodo4 = {
+      id: '4',
+      descrition: 'task5',
+      completed: false,
+      index: 4,
+    };
+    todoList.add(newTodo4);
+    todoList.changeStatus(newTodo4.id, true);
+    expect(todoList.tasksArr[3].completed).toBeTruthy();
+    expect(todoList.tasksArr).toHaveLength(4);
+  });
+});
+
+describe('Clear all completed', () => {
+  test('Clear completed items', () => {
+    const todoList = new TasksBluePrint();
+    todoList.removeCompletedTasks();
+    expect(todoList.tasksArr).toHaveLength(3);
```

Comment:
- [ ] kindly clarify why you are expecting three tasks to remain after clearing all completed? In case the test cases are cascaded, I suggest that each test case should self container. That is, it should not depend on other test case results.

Reply:
@gtekle I am expecting them to remain because they're not completed.